core/src/main/java/io/grpc/internal/AbstractServerImplBuilder.java

```diff
@@ -108,6 +111,23 @@ public final T addService(ServerServiceDefinition service) {
     throw new UnsupportedOperationException("Underlying HandlerRegistry is not mutable");
   }
 
+  /**
+   * Adds a service implementation to the handler registry.
+   *
+   * <p>This is supported only if the user didn't provide a handler registry, or the provided one is
+   * a {@link MutableHandlerRegistry}. Otherwise it throws an UnsupportedOperationException.
+   *
+   * @param bindableService BindableService object.
+   */
+  @Override
```

Comment:
`@ExperimentalApi` for transitivity. I expect we'll the annotation after this impending release.